vaadin-upload-flow-parent/vaadin-upload-flow-integration-tests/src/test/java/com/vaadin/flow/component/upload/tests/UploadInterruptIT.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2000-2025 Vaadin Ltd.
+ *
+ * This program is available under Vaadin Commercial License and Service Terms.
+ *
+ * See  {@literal <https://vaadin.com/commercial-license-and-service-terms>}  for the full
+ * license.
+ */
+package com.vaadin.flow.component.upload.tests;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.WebElement;
+
+import com.vaadin.flow.component.upload.testbench.UploadElement;
+import com.vaadin.flow.testutil.TestPath;
+
+@TestPath("vaadin-upload/interrupt")
+public class UploadInterruptIT extends AbstractUploadIT {
+
+    private UploadElement upload;
+    private WebElement uploadOutput;
+    private WebElement eventsOutput;
+
+    @Before
+    public void init() {
+        open();
+        upload = $(UploadElement.class).waitForFirst();
+        uploadOutput = $("div").id("test-output");
+        eventsOutput = $("div").id("test-events-output");
+    }
+
+    @Test
+    public void uploadMultipleFiles_interruptUpload_allOngoingUploadsInterrupted()
+            throws Exception {
+
+        List<File> files = new ArrayList<>();
+        files.add(createTempFile("txt"));
+        files.add(createTempFile("txt"));
+        files.add(createTempFile("interrupt.txt"));
+        files.add(createTempFile("txt"));
+        files.add(createTempFile("txt"));
+
+        upload.setProperty("noAuto", true);
+        files.forEach(file -> upload.upload(file, 0));
+        executeScript("arguments[0].uploadFiles()", upload);
+        waitForUploads();
```

Comment:
This required copying some logic from newer versions, as the `uploadMultiple` method does not exist in v14.